### PR TITLE
:bug: Fix some SM bugs

### PR DESCRIPTION
**Pull request by ezrasilvera**
## Related issue(s)

Fixes #

### CODE REVIEW

File: space-framework/pkg/space-manager/reconcile_spaceprovider.go [hunk 1]
```diff
@@ -51,9 +51,8 @@ func (c *controller) reconcileSpaceProviderDesc(key string) error {
 }
 
 func (c *controller) handleAdd(providerDesc *spacev1alpha1.SpaceProviderDesc) error {
-	if providerDesc.Status.Phase == spacev1alpha1.SpaceProviderDescPhaseInitializing {
```
Comment (andreyod) [inline space-framework/pkg/space-manager/reconcile_spaceprovider.go:54]: So what will happen with the event when provider status phase is Initializing? We will try to set it again to Initializing and create a provider?

Comment (ezrasilvera) [reply space-framework/pkg/space-manager/reconcile_spaceprovider.go:54]: yes. I went through the code and it seems our Create flow is idempotent and doesn't duplicate things

**Pull request merged**